src/server/hook_func.c

```diff
@@ -4305,7 +4305,7 @@ int server_process_hooks(int rq_type, char *rq_user, char *rq_host, hook *phook,
 	*num_run += 1;
 	if (pbs_python_get_scheduler_restart_cycle_flag() == TRUE) {
 
-		set_scheduler_flag(SCH_SCHEDULE_RESTART_CYCLE);
+		set_scheduler_flag(SCH_SCHEDULE_RESTART_CYCLE, dflt_scheduler);
```

Comment:
We will need to know which scheduler's sched cycle needs to be restarted. I guess this comment does not apply to the current pull but it needs to be addressed in future.

Reply:
@arungrover 
What you observed is right. This does not apply to the current pull request where as we can address this as part of subsequent interfaces. For the time-being only default scheduler is restarted from the hooks interface.